plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/s3/ByteBufferAsyncRequestBody.java

```diff
@@ -1,116 +0,0 @@
-/*
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package io.trino.plugin.exchange.filesystem.s3;
-
-import io.airlift.log.Logger;
-import org.reactivestreams.Subscriber;
-import org.reactivestreams.Subscription;
-import software.amazon.awssdk.core.async.AsyncRequestBody;
-import software.amazon.awssdk.core.internal.async.ByteArrayAsyncRequestBody;
-import software.amazon.awssdk.core.internal.util.Mimetype;
-
-import java.nio.ByteBuffer;
-import java.util.Optional;
-
-import static java.util.Objects.requireNonNull;
-
-/**
- * This class mimics the implementation of {@link ByteArrayAsyncRequestBody} except for we use a ByteBuffer
```

Comment:
This was here for a reason

Reply:
cc: @linzebing

Reply:
Now when you changed `fromByteBufferUnsafe` it should be fine. 
@linzebing PTAL.